third_party/2and3/pytz/__init__.pyi

```diff
@@ -23,4 +23,20 @@ class _UTCclass(dt.tzinfo):
 utc = ...  # type: _UTCclass
 UTC = ...  # type: _UTCclass
 
-def timezone(zone: str) -> dt.tzinfo: ...
+
+class _BaseTzInfo(dt.tzinfo):
+    zone = ...  # type: str
+
+    def fromutc(self, dt: dt.datetime) -> dt.datetime: ...
+    def localize(self, dt: dt.datetime, is_dst: Optional[bool]=False) -> dt.datetime: ...
+
+
+class _StaticTzInfo(_BaseTzInfo):
+    def normalize(self, dt: dt.datetime, is_dst: Optional[bool]=False) -> dt.datetime: ...
+
+
+class _DstTzInfo(_BaseTzInfo):
+    def normalize(self, dt: dt.datetime) -> dt.datetime: ...
+
+
+def timezone(zone: str) -> Union[_StaticTzInfo, _DstTzInfo]: ...
```

Comment:
Sadly you (currently) cannot return something that uses a TypeVar that does not occur in the arguments.

Maybe it works to define normalize() without the optional flag on the common base class and then override it in StaticTzInfo with the optional flag added -- that's acceptable for the subclassing type checks, and then users who want to call normalize() without the is_dst flag won't have to worry about the union. Users who do want to call it with the is_dst flag can still use `if isinstance(tz, StaticTzInfo): tz.normalize(dt, is_dst=FLAG)` -- which technically they'd have to do anyways.